nixos/modules/services/x11/xbanish.nix

```diff
@@ -0,0 +1,35 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let cfg = config.services.xbanish;
+
+in {
+  options.services.xbanish = {
+
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      example = true;
+      description = "Enable xbanish to hide your mouse pointer while not in use";
+    };
+
+    arguments = mkOption {
+      description = "Arguments to pass to xbanish command";
+      default = "";
+      example = "-d -i shift";
+      type = types.str;
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.user.services.xbanish = {
+      description = "xbanish";
+      wantedBy = [ "default.target" ];
```

Comment:
ah now i understand - user services are isolated from system services so there is no way for this to depend on `graphical.target`. that explains why all these services have `serviceConfig.Restart: "always"` 

i can hold off on this PR and figure out a better approach or i can move forward and open a separate issue, which would you prefer?

i'll also fix the description when i get home :) (i'm following https://nixos.org/wiki/Pull_request for the format of the commit message, would you like me to add a multiline message which includes the service description?)